File: iOS/MarigoldStreamExamples/MarigoldStreamExamples/GraphicalCardTableViewCell.m
//
//  GraphicalCardTableViewCell.m
//  MarigoldStreamExamples
//
//  Created by Sam Jarman on 9/09/15.
//  Copyright (c) 2015 Marigold Mobile. All rights reserved.
//

#import "GraphicalCardTableViewCell.h"
#import <SDWebImage/UIImageView+WebCache.h>

@interface GraphicalCardTableViewCell ()
@property (weak, nonatomic) IBOutlet UIView *timeBackground;

@end

@implementation GraphicalCardTableViewCell

- (void)configureCellWithMessage:(MARMessage *)message {
    [super configureCellWithMessage:message];
    [self configureTimeBackground];
    if (message.imageURL) {
        [self.imgView sd_setImageWithURL:message.imageURL placeholderImage:[UIImage imageNamed:@"placeholder_image"]];
        self.imgView.contentMode = UIViewContentModeScaleAspectFill;
        self.imgView.clipsToBounds = YES;
    }
}

+ (NSString *)cellIndentifier {
    return @"GraphicalCardTableViewCell";
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    UIColor *backgroundColor = self.timeBackground.backgroundColor;
    [super setSelected:selected animated:animated];
    self.timeBackground.backgroundColor = backgroundColor;
}

- (void)setHighlighted:(BOOL)highlighted animated:(BOOL)animated {
    UIColor *backgroundColor = self.timeBackground.backgroundColor;
    [super setHighlighted:highlighted animated:animated];
    self.timeBackground.backgroundColor = backgroundColor;
}

- (void)configureTimeBackground {
    self.timeBackground.layer.cornerRadius = 4.0;
}

@end

File: iOS/MarigoldSwiftStreamExamples/MarigoldSwiftStreamExamples/GraphicalCardTableViewCell.swift
//
//  GraphicalCardTableViewCell.swift
//  MarigoldSwiftStreamExamples
//
//  Created by Sam Jarman on 14/09/15.
//  Copyright (c) 2015 Marigold Mobile. All rights reserved.
//

import UIKit

class GraphicalCardTableViewCell: TextCardTableViewCell {
    @IBOutlet weak var imgView: UIImageView!
    @IBOutlet weak var timeBackground: UIView!
    
    override class func cellIdentifier() -> String {
        return "GraphicalCardTableViewCell"
    }
    
    override func configureCell(message: MARMessage) {
        super.configureCell(message: message)
        if message.imageURL != nil {
            self.imgView.sd_setImage(with: message.imageURL, placeholderImage: UIImage(named: "placeholder_image"))
            self.imgView.contentMode = UIView.ContentMode.scaleAspectFill
            self.imgView.clipsToBounds = true
        }
        self.timeBackground.layer.cornerRadius = 4.0
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        let backgroundColor = self.unreadLabel.backgroundColor
        super.setSelected(selected, animated: animated)
        self.unreadLabel.backgroundColor = backgroundColor
    }
    
    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        let backgroundColor = self.unreadLabel.backgroundColor
        super.setHighlighted(isSelected, animated: animated)
        self.unreadLabel.backgroundColor = backgroundColor
    }
}
